pyrival/algebra/mod_template.py

```diff
@@ -0,0 +1,81 @@
+"""
+This template is useful for problems involving a prime modulo. 
+The template contains a fast function for calculating a * b % MOD, 
+as well as precalculating factorial, inverse factorial, modular inverse
+for all integers < maxN in O(maxN) time.
+
+With this template, one can for example quickly calculate n choose k mod MOD,
+or calculate matrix multiplication mod MOD.
+"""
+
+def fast_modder(MOD):
+    """ Returns function modmul(a,b) that quickly calculates a * b % MOD, assuming 0 <= a,b < MOD """
+    import sys, platform
```

Comment:
I'm tempted to move this up, to the top of the file, even though this is specific.